lib/fixture_factory/definition.rb

```diff
@@ -5,14 +5,14 @@ class Definition # :nodoc:
     EMPTY_BLOCK = proc {}
 
     attr_writer   :block
-    attr_accessor :class_name, :fixture_method, :fixture_name, :parent, :sequence
+    attr_accessor :fixture_method, :fixture_name, :parent, :sequence, :lazy_class
 
     def initialize(name, options = {})
-      self.parent         = options.fetch(:parent)     { default_parent_for(name) }
-      self.class_name     = options.fetch(:class_name) { parent.class_name }
```

Comment:
This is a breaking change. If we are to support both cases we can't remove the use of this option.

Reply:
If you look into the registry file

Reply:
[If you look into the `lib/fixture_factory/registry.rb` file](https://github.com/Shopify/fixture_factory/pull/8/files#diff-0f03a0536cc08da8a88bf6d86bae8041a1a298c395fe9b35e4fc475ffe12dc1f), you can see that we are keeping backward compatibility by wrapping the string constant. Then the option gets renamed so that we pass it into the `Definition` class. That way, we can reduce the complexity of the `Definition` class so that it only needs to know how to handle blocks rather than two different approaches. 

```ruby
      def factory(name, options = {}, &block)
        if options.key?(:class)
          options[:class] ||= -> { options.delete(:class_name).to_s.constantize }
```

Reply:
My mistake, I expected it to be in the definition. 👍